cmd/encryption-v1.go

```diff
@@ -385,22 +363,73 @@ func newEncryptReader(content io.Reader, key []byte, bucket, object string, meta
 
 	return reader, nil
 }
+func generateKMSKey(metadata map[string]string) ([]byte, error) {
+	if !globalIsVaultEnabled {
```

Comment:
Not needed. My intention was:
You have two global values:
1. `globalKMSKeyID string`
2. `globalKMS crypto.KMS`

If `globalKMS == nil` -> error KMS not configured
`key, encKey, err := globalKMS.GenerateKey(globalKMSKeyID, ctx)`

Also the context should be filled with some values. At least bucket, object name. I'm also thinking of some other informations... For now bucket, object name is fine but we have to specify *before* merging.